Building on Windows 11 and Visual Studio 2022

I'm trying to build the commit `110a5be03db6abd4bb4fe9b6d4a70997c4e3a656` with _Visual Studio 2022_ under _Windows 11_. Unfortunately, I can't do that. I got error with very legacy dependency of _.NET 4.5_.

```txt
D:\Programs\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(12
17,5): error MSB3644: The reference assemblies for .NETFramework,Version=v4.5 were not found. To resolve this, install
the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET
Framework Developer Packs at https://aka.ms/msbuild/developerpacks [D:\Development\DirectXShaderCompiler\build.x64\tool
s\clang\tools\dotnetc\dndxc.csproj]
```

I think it's not a good idea to require such legacy tool in 2021. For the record, I tried to install _.NET 4.5.1_ development pack. There is no _.NET 4.5_ anymore. Look [here](https://docs.microsoft.com/en-us/dotnet/framework/install/guide-for-developers).

![image](https://user-images.githubusercontent.com/6526711/141641197-554bafcf-41ff-486a-a188-0bb295990e3d.png)


I used the following building commands:

```txt
cd /D D:\Development\DirectXShaderCompiler
git submodule update --init
.\utils\hct\hctstart.cmd -x64 D:\Development\DirectXShaderCompiler D:\Development\DirectXShaderCompiler\build.x64
.\utils\hct\hctbuild.cmd -official -rel -x64 -vs2022 -speak-up -no-dxilconv -spirv
```

For the record previously I was able to build _DXC_ project under _Windows 10_ and _Visual Studio 2019_. But for some reason trying to build the project from fresh sources is impossible with _Win11_ and _VS2022_. The _Visual Studio 2022_ is officially supported according to `.\utils\hct\hctbuild.cmd`. Something is missing in documentation of the repo? Would you kindly help?

My build environment is the following:

- [_Windows 11 WDK_](https://go.microsoft.com/fwlink/?linkid=2166289)
- _Windows 11 Pro 64-bit_ (10.0, Build 22000)
- _Visual Studio 2022 Community_
-  Desktop development with C++
- Universal Windows Platform development
- _.NET_ 6.0 Runtime
- _.NET Native_
- _.NET SDK_
- _CLR_ data types for _SQL Server_ (**is it required??**)
- Developer Analytics tools
- _NuGet_ package manager
- Text Template Transformation
- _C#_ and Visual Basic Roslyn compilers
- _C++_ 2022 Redistributable Update
- _C++_ CMake tools for _Windows_
- _MSBuild_
- _MSVC v143 - VS 2022 C++ x64/x86 build tools (Latest)_
- _Windows_ Universal _CRT SDK_
- _.NET_ profiling tools
- _C++ AddressSanitizer_
- _C#_ and _Visual Basic_
- _C++_ Build Tools core features
- _IntelliCode_
- Image and _3D_ model editors (**is it required??**)
- _C++ ATL_ for latest v143 build tools (x86 & x64) (**also legacy in my opinion**)
- _Windows 10 SDK_ (10.0.19041.0) 
- _Windows 11 SDK_ (10.0.22000.0)
- _Windows_ Universal C Runtime

The error log is the following:

```txt
C:\Users\Goshido>cd /D D:\Development\DirectXShaderCompiler

D:\Development\DirectXShaderCompiler>.\utils\hct\hctstart.cmd -x64 D:\Development\DirectXShaderCompiler D:\Development\DirectXShaderCompiler\build.x64
Default architecture - set BUILD_ARCH=x64
HLSL source directory set to HLSL_SRC_DIR=D:\Development\DirectXShaderCompiler
HLSL build directory set to HLSL_BLD_DIR=D:\Development\DirectXShaderCompiler\build.x64

You can recreate the environment with this command.
.\utils\hct\hctstart.cmd -x64 D:\Development\DirectXShaderCompiler D:\Development\DirectXShaderCompiler\build.x64

Setting up macros for this console - run hcthelp for a reference.

Found Windows 10 SDK 10.0.22000.0
Path adjusted to include TAEF te.exe.
Path adjusted to include git.

D:\Development\DirectXShaderCompiler>.\utils\hct\hctbuild.cmd -official -rel -x64 -vs2022 -speak-up -no-dxilconv -spirv
Will generate official version for build
SPIR-V codegen is enabled.
Creating solution files for x64, logging to D:\Development\DirectXShaderCompiler\build.x64\cmake-log.txt
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Building DxcDisassembler.inc...
  Updating DxcDisassembler.inc...
  Building DxcOptimizer.inc...
  Updating DxcOptimizer.inc...
  Building DxilConstants.h...
  Building docs/DXIL.rst...
  MSFileSystemImpl.cpp
  Building DxilInstructions.h...
  Building gen_intrin_main_tables_15.h...
  Building DxilCounters.h...
  Building DxilMetadataHelper.cpp...
  Error.cpp
  Main.cpp
  Record.cpp
  SetTheory.cpp
  StringMatcher.cpp
  TableGenBackend.cpp
  TGLexer.cpp
  TGParser.cpp
  Building DxilSigPoint.inl...
  Building DxilPIXPasses.inc...
  Building DxilShaderModel.cpp...
  Updating gen_intrin_main_tables_15.h...
  Generating version.inc.gen
  Generating version
  Building DxilShaderModel.h...
  Building DxilOperations.cpp...
  Building DxilValidationImpl.inc...
  Updating DxilPIXPasses.inc...
  Updating DxilValidationImpl.inc...
  Building HlslIntrinsicOp.h...
  Building HLSLOptions.td...
D:\Programs\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(12
17,5): error MSB3644: The reference assemblies for .NETFramework,Version=v4.5 were not found. To resolve this, install
the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET
Framework Developer Packs at https://aka.ms/msbuild/developerpacks [D:\Development\DirectXShaderCompiler\build.x64\tool
s\clang\tools\dotnetc\dndxc.csproj]
  Touch GetCommitInfo.py to trigger rebuild
  Building DxilValidation.inc...
  Generating dxcversion.inc.gen
  Generating version
  LLVMMSSupport.vcxproj -> D:\Development\DirectXShaderCompiler\build.x64\Release\lib\LLVMMSSupport.lib
  Updating DxilValidation.inc...
  Building instrumentation manifest ...
  Updating instrumentation manifest ...
  APFloat.cpp
  APInt.cpp
  APSInt.cpp
  ARMBuildAttrs.cpp
  ARMWinEH.cpp
  Allocator.cpp
  BlockFrequency.cpp
  BranchProbability.cpp
  circular_raw_ostream.cpp
  COM.cpp
  CommandLine.cpp
  Compression.cpp
  ConvertUTFWrapper.cpp
  CrashRecoveryContext.cpp
  DataExtractor.cpp
  DataStream.cpp
Attempting to cancel the build...
^CTerminate batch job (Y/N)? y

D:\Development\DirectXShaderCompiler>
```

--- a/tools/clang/tools/dotnetc/dndxc.csproj.txt
+++ b/tools/clang/tools/dotnetc/dndxc.csproj.txt
@@ -9,7 +9,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>MainNs</RootNamespace>
     <AssemblyName>dndxc</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <ProjectTypeGuids>{60dc8134-eba5-43b8-bcc9-bb4bc16c2548};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
     <WarningLevel>4</WarningLevel>

